src/librustdoc/clean/types.rs

```diff
@@ -676,15 +676,23 @@ impl Item {
         }
         let header = match *self.kind {
             ItemKind::ForeignFunctionItem(_) => {
-                let abi = tcx.fn_sig(self.item_id.as_def_id().unwrap()).abi();
+                let def_id = self.item_id.as_def_id().unwrap();
+                let abi = tcx.fn_sig(def_id).abi();
                 hir::FnHeader {
                     unsafety: if abi == Abi::RustIntrinsic {
                         intrinsic_operation_unsafety(tcx, self.item_id.as_def_id().unwrap())
                     } else {
                         hir::Unsafety::Unsafe
                     },
                     abi,
-                    constness: hir::Constness::NotConst,
+                    constness: if abi == Abi::RustIntrinsic
```

Comment:
Just realized it but: why the `abi` check?

Reply:
Mostly to mirror the above check. I'm assuming that "rust-intrinsic" is the only kind of `ForeignFunctionItem` which can be const so it seemed best to be conservative and only do the const checks for that kind.

Reply:
Makes sense!